main.cpp

```diff
@@ -49,77 +49,105 @@ Create a branch named Part1
+        std::cout << "Reducing magnitude.. \n";
+        while( std::abs (static_cast <long> (this->var2 - this->var1)) > 0.001f )
+        {
+            /*
+             write something that makes the distance between that->name2 and that->name1 get smaller
+             */
+            this->var2 += 0.5f;
+            std::cout << this->var2 << ", ";
+
+        }
+        std::cout << "\n(non-static) U's var2 updated value: " << this->var2 << std::endl;
+        return this->var2 * this->var1;
     }
 };
 
-struct <#structname2#>
+struct StaticTransform
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float product(U* that, float* inputValue ) 
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if ( inputValue != nullptr && that != nullptr)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout << "\nU's var1 value: " << that->var1 << std::endl;
+            that->var1 = *inputValue; 
+            std::cout << "\nU's var1 updated value: " << that->var1 << std::endl;
+            std::cout << "Reducing magnitude.. \n";
+            while( std::abs (static_cast <long> (that->var2 - that->var1)) > 0.001f )
+                {
+                    /*
+                    write something that makes the distance between that->name2 and that->name1 get smaller
+                     */          
+                    that->var2 += 0.5f;
+                    std::cout << that->var2 << ", ";
+                }
+        std::cout << "\nU's var2 updated value: " << that->var2 << std::endl;
+        return (that->var2 * that->var1);
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        return *inputValue; 
     }
 };
         
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+
+    T var1( 0.1f , "a" );                                             
+    T var2( 0.5f , "b");  
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    Operation f;                                      
+    T* smaller = f.compare( &var1, &var2 );  
+    if (smaller != nullptr) std::cout << "\n\nthe smaller one is << " << smaller->name << std::endl; 
+    else std::cout << "Something is wrong with the compare function..." << std::endl; 
```

Comment:
This line is technically not needed as the instructions never ask you to add an `else { }` block, but since you added it:
```
else they have the same value or one of the inputs is nullptr!   
```
With this project, I'm hoping you'll look at the function implementations and understand the behaviors of the code that's written.